File: backend/src/main/java/lan/dk/podcastserver/manager/worker/Extractor.java
package lan.dk.podcastserver.manager.worker;

import lan.dk.podcastserver.entity.Item;
import lan.dk.podcastserver.manager.downloader.DownloadingItem;
import org.apache.commons.io.FilenameUtils;
import org.apache.commons.lang3.StringUtils;

import static io.vavr.API.Some;

/**
 * Created by kevin on 03/12/2017
 */
public interface Extractor {

    DownloadingItem extract(Item item);
    Integer compatibility(String url);

    default String getFileName(Item item) {
        return Some(item.getUrl())
                .map(s -> StringUtils.substringBefore(s, "?"))
                .map(FilenameUtils::getName)
                .getOrElse("");
    };
}

File: backend/src/main/kotlin/com/github/davinkevin/podcastserver/manager/worker/Extractor.kt
package com.github.davinkevin.podcastserver.manager.worker

import com.github.davinkevin.podcastserver.manager.downloader.DownloadingInformation
import com.github.davinkevin.podcastserver.manager.downloader.DownloadingItem
import org.apache.commons.io.FilenameUtils
import java.net.URI

/**
 * Created by kevin on 03/12/2017
 */
interface Extractor {

    fun extract(item: DownloadingItem): DownloadingInformation
    fun compatibility(url: URI): Int

    fun getFileName(url: URI): String {
        val urlFile =  url.toASCIIString().substringBefore("?")
        return FilenameUtils.getName(urlFile)
    }
}
